### PR TITLE
Error for missing clustername on container

### PR DESCRIPTION
**Pull request by agarakan**
# Description of the issue
Pulling in missing error logic when clusterName is missing on a container. This is an edge case that we should ideally not see but should still log errors for. 

# Description of changes
Add error check and test

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Added and ran unit tests
```
make lint
make fmt
make fmt-sh
make test
```
# Requirements
_Before commiting your code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

-------
### Integration Tests
To run integration tests against this PR, add the `ready for testing` label.

### CODE REVIEW

File: translator/translate/otel/receiver/prometheus/translator.go [hunk 1]
```diff
@@ -238,6 +240,11 @@ func getClusterNameFromConfig(conf *confmap.Conf, promConfigKey string) string {
 		clusterName = util.GetECSClusterName(clusterNameSectionKey, prometheusConfigInput)
 	}
 
+	// Cluster Name is mandatory for Containerized Workloads
+	if context.CurrentContext().RunInContainer() && clusterName == "" {
+		translator.AddErrorMessages(promConfigKey, "ClusterName is not defined")
+	}
+
```
Comment (musa-asad) [inline translator/translate/otel/receiver/prometheus/translator.go:247]: Not sure if this is viable, but is there any way we can consolidate the logic here in the `ApplyRule` and call that?

Comment (agarakan) [reply translator/translate/otel/receiver/prometheus/translator.go:247]: actually this branch removes the applyRule methods to just process the config linearly.

**Review by musa-asad — APPROVED** [review]

**Review by TravisStark — APPROVED** [review]
